lib/matplotlib/constrained_layout.py

```diff
@@ -0,0 +1,221 @@
+from __future__ import division, print_function
+
+from matplotlib.legend import Legend
+import matplotlib.transforms as transforms
+import numpy as np
+import matplotlib.layoutbox as layoutbox
+
+
+#########
+def  get_axall_tightbbox(ax, renderer):
+
+
+
+    # main bbox of the axis....
+    bbox = ax.get_tightbbox(renderer=renderer)
+    # now add the possibility of the legend...
+    for child in ax.get_children():
+        if isinstance(child, Legend):
+            bboxn = child._legend_box.get_window_extent(renderer)
+            bbox = transforms.Bbox.union([bbox, bboxn])
```

Comment:
I think this would be more specific in sniffing out the legend type stuff.

```python
possible_legends = ax.artists + [ax.legend_] if ax.legend_ else []
for child in possible_legends:
    if isinstance(child, (Legend, AnchoredOffsetbox)):
        ...

```

However, overall this is a neat little function that can bundle up all types of adornments to the axes (at the risk of complicating it).